dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb

```diff
@@ -0,0 +1,366 @@
+    @admin = create(:admin)
+    @organizer = create(:workshop_organizer)
+    @facilitator = create(:facilitator)
+
+    @workshop = create(:pd_workshop, organizer: @organizer, facilitators: [@facilitator])
+    @standalone_workshop = create(:pd_workshop)
+  end
+
+  # Action: Index
+
+  test 'admins can list all workshops' do
+    sign_in @admin
+    assert_equal 2, Pd::Workshop.count
+
+    get :index
+    assert_response :success
+    assert_equal 2, JSON.parse(@response.body).length
+  end
+
+  test 'workshop organizers can list all their workshops' do
+    sign_in @organizer
+    get :index
+    assert_response :success
+    assert_equal 1, JSON.parse(@response.body).length
+  end
+
+  test 'workshop organizers cannot list workshops they are not organizing' do
+    sign_in create(:workshop_organizer)
+    get :index
+    assert_response :success
+    assert_equal 0, JSON.parse(@response.body).length
+  end
+
+  test 'facilitators can list workshops they are facilitating' do
+    sign_in @facilitator
+    get :index
+    assert_response :success
+    assert_equal 1, JSON.parse(@response.body).length
+    sign_out @facilitator
+  end
+
+  test 'facilitators cannot list workshops they are not facilitating' do
+    sign_in create(:facilitator)
+    get :index
+    assert_response :success
+    assert_equal 0, JSON.parse(@response.body).length
+  end
+
+  test 'filter by state' do
+    sign_in @admin
+    workshop_in_progress = create :pd_workshop
+    workshop_in_progress.sessions << create(:pd_session)
+    workshop_in_progress.start!
+    assert_equal Pd::Workshop::STATE_IN_PROGRESS, workshop_in_progress.state
+
+    get :index, state: Pd::Workshop::STATE_IN_PROGRESS
+    assert_response :success
+    response = JSON.parse(@response.body)
+    assert_equal 1, response.length
+    assert_equal workshop_in_progress.id, response[0]['id']
+  end
+
+  # Action: Show
+
+  test 'admins can view workshops' do
+    sign_in @admin
+    get :show, id: @workshop.id
+    assert_response :success
+    assert_equal @workshop.id, JSON.parse(@response.body)['id']
+  end
+
+  test 'workshop organizers can view their workshops' do
+    sign_in @organizer
+    get :show, id: @workshop.id
+    assert_response :success
+    assert_equal @workshop.id, JSON.parse(@response.body)['id']
+  end
+
+  test 'workshop organizers cannot view a workshop they are not organizing' do
+    sign_in @organizer
+    get :show, id: @standalone_workshop.id
+    assert_response :forbidden
+  end
+
+  test 'facilitators can view a workshop they are facilitating' do
+    sign_in @facilitator
+    get :show, id: @workshop
+    assert_response :success
+    assert_equal @workshop.id, JSON.parse(@response.body)['id']
+  end
+
+  test 'facilitators cannot view a workshop they are not facilitating' do
+    sign_in @facilitator
+    get :show, id: @standalone_workshop
+    assert_response :forbidden
+  end
+
+  # Action: Create
+
+  test 'admins can create workshops' do
+    sign_in @admin
```

Comment:
Combine these two - you can use the assert_difference or assert_creates method both times.